frame/nomination-pools/src/tests.rs

```diff
@@ -5100,3 +5215,1045 @@ mod reward_counter_precision {
+					1,
+					Some((Perbill::from_percent(10), 900))
+				),
+				Error::<Runtime>::CommissionChangeThrottled
+			);
+
+			// Run to block 3
+			run_blocks(2);
+
+			// Now try to increase commission by 1% and provide an initial payee. This should
+			// succeed and set the `throttle_from` field.
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(6), 900))
+			));
+			assert_eq!(
+				BondedPool::<Runtime>::get(1).unwrap().commission,
+				Commission {
+					current: Some((Perbill::from_percent(6), 900)),
+					max: None,
+					change_rate: Some(CommissionChangeRate {
+						max_increase: Perbill::from_percent(1),
+						min_delay: 2_u64
+					}),
+					throttle_from: Some(3_u64),
+				}
+			);
+
+			// Attempt to increase the commission an additional 1% (now 7%). this will fail as
+			// `throttle_from` is now the current block. At least 2 blocks need to pass before we
+			// can set commission again.
+			assert_noop!(
+				Pools::set_commission(
+					RuntimeOrigin::signed(900),
+					1,
+					Some((Perbill::from_percent(7), 900))
+				),
+				Error::<Runtime>::CommissionChangeThrottled
+			);
+
+			// Run 2 blocks into the future, to block 3.
+			run_blocks(2);
+
+			// Can now successfully increase the commission again, to 7%.
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(7), 900)),
+			));
+
+			// Run 2 blocks into the future, to block 5.
+			run_blocks(2);
+
+			// Now surpassed the `min_delay` threshold, but the `max_increase` threshold is
+			// still at play. An attempted commission change now to 8% (+2% increase) should fail.
+			assert_noop!(
+				Pools::set_commission(
+					RuntimeOrigin::signed(900),
+					1,
+					Some((Perbill::from_percent(9), 900)),
+				),
+				Error::<Runtime>::CommissionChangeThrottled
+			);
+
+			// Now set a max commission to the current 5%. This will also update the current
+			// commission to 5%.
+			assert_ok!(Pools::set_commission_max(
+				RuntimeOrigin::signed(900),
+				1,
+				Perbill::from_percent(5)
+			));
+			assert_eq!(
+				BondedPool::<Runtime>::get(1).unwrap().commission,
+				Commission {
+					current: Some((Perbill::from_percent(5), 900)),
+					max: Some(Perbill::from_percent(5)),
+					change_rate: Some(CommissionChangeRate {
+						max_increase: Perbill::from_percent(1),
+						min_delay: 2
+					}),
+					throttle_from: Some(7)
+				}
+			);
+
+			// Run 2 blocks into the future so we are eligible to update commission again.
+			run_blocks(2);
+
+			// Now attempt again to increase the commission by 1%, to 6%. This is within the
+			// change rate allowance, but `max_commission` will now prevent us from going any
+			// higher.
+			assert_noop!(
+				Pools::set_commission(
+					RuntimeOrigin::signed(900),
+					1,
+					Some((Perbill::from_percent(6), 900)),
+				),
+				Error::<Runtime>::CommissionExceedsMaximum
+			);
+
+			assert_eq!(
```

Comment:
ditto, I think checking events at the end of the tests is reasonable when the tests is by all means a "unti" tests, testing one scenario. Your tests tend to me much longer, and at least I find it hard to connect the events with everything that happened in the test.

Reply:
Okay, will amend.